ArcGISRuntimeSDKQt_QMLSamples/Layers/DisplaySubtypeFeatureLayer/DisplaySubtypeFeatureLayer.qml

```diff
@@ -0,0 +1,192 @@
+// [WriteFile Name=DisplaySubtypeFeatureLayer, Category=Layers]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.6
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import QtQuick.Shapes 1.12
+import Esri.ArcGISRuntime 100.8
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    readonly property var labelJson: { "labelExpression": "[nominalvoltage]", "labelPlacement": "esriServerPointLabelPlacementAboveRight", "useCodedValues": true, "symbol": { "angle": 0, "backgroundColor": [ 0, 0, 0, 0 ], "borderLineColor": [ 0, 0, 0, 0 ], "borderLineSize": 0, "color": [ 0, 0, 255, 255 ], "font": { "decoration": "none", "size": 10.5, "style": "normal", "weight": "normal" }, "haloColor": [ 255, 255, 255, 255 ], "haloSize": 2, "horizontalAlignment": "center", "kerning": false, "type": "esriTS", "verticalAlignment": "middle", "xoffset": 0, "yoffset": 0 } }
+    property var subtypeSublayer
+    property var originalRenderer
+    property var mapScale: mapView ? Math.round(mapView.mapScale) : 0
+    property var sublayerMinScale
+
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Map {
+            BasemapStreetsNightVector {}
+
+            // create the feature layer
+            SubtypeFeatureLayer  {
+                id: subtypeFeatureLayer
+
+                // feature table
+                ServiceFeatureTable {
+                    id: featureTable
+                    url: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/100"
+                }
+
+                onLoadStatusChanged: {
+                    if (loadStatus != Enums.LoadStatusLoaded)
+                        return;
+
+                    // get the Street Light sublayer and define its labels
+                    subtypeSublayer = subtypeFeatureLayer.sublayerWithSubtypeName("Street Light");
+                    var labelDefinition = ArcGISRuntimeEnvironment.createObject("LabelDefinition", { json : labelJson});
+
+                    if (!labelDefinition || !subtypeSublayer)
+                      return;
+
+                    subtypeSublayer.labelDefinitions.append(labelDefinition);
+                    subtypeSublayer.labelsEnabled = true;
+
+                    // get the original renderer of the sublayer
+                    originalRenderer = subtypeSublayer.renderer;
```

Comment:
I would think since this is not needed in Qml, it is not needed for C++, either.

Reply:
Not sure what you mean here. We need the original renderer to be preserved because you can switch between the default renderer and the alternate renderer in the sample. Once you apply the alternate renderer we will lose the default and won't be able to switch back. Is my understanding of this correct?

Reply:
Oops, poorly written comment.

I was referring to the clone method used in C++. Since clone is not needed here, that should be an argument that clone is not needed in C++, either.

Reply:
Ah I see what you mean now. yeah it works just fine without the clone. Thank you for the added clarification on when clone should be used!